### PR TITLE
build: use latest manylinux images

### LINKED ISSUE #1304: No py3.8 manylinux wheels
Can this project start publishing py38 manylinux wheels? The [package says](https://github.com/DataDog/dd-trace-py/blob/d14bfec34db3d8cd9544863b1b444746fcd369ee/setup.py#L132) it supports it. And it looks like [CI is configured](https://github.com/DataDog/dd-trace-py/blob/d14bfec34db3d8cd9544863b1b444746fcd369ee/.circleci/config.yml#L168) to test it. However, there aren't any [wheels published for py38](https://pypi.org/project/ddtrace/0.35.0/#files).

### PR DESCRIPTION
**Pull request by majorgreys**
Fixes #1304

### CODE REVIEW

File: scripts/build-dist [hunk 1]
```diff
@@ -40,10 +40,12 @@ EOF
 python setup.py sdist --dist-dir "$BUILD_DIR"
 
 # Build x86_64 linux and manylinux wheels
-docker run -it --rm -v "${BUILD_DIR}:/dd-trace-py" -e "ARCH=x86_64" quay.io/pypa/manylinux1_x86_64 /bin/bash -c "${build_script}"
+docker pull quay.io/pypa/manylinux1_x86_64
+docker run -it --rm -v "${BUILD_DIR}:/dd-trace-py" -e "ARCH=x86_64" quay.io/pypa/manylinux1_x86_64:latest /bin/bash -c "${build_script}"
```
Comment (brettlangdon) [inline scripts/build-dist]:
```suggestion
docker pull quay.io/pypa/manylinux1_x86_64:latest
docker run -it --rm -v "${BUILD_DIR}:/dd-trace-py" -e "ARCH=x86_64" quay.io/pypa/manylinux1_x86_64:latest /bin/bash -c "${build_script}"
```

File: scripts/build-dist [hunk 2]
```diff
@@ -40,10 +40,12 @@ EOF
...
 
 # Build i686 linux and manylinux wheels
-docker run -it --rm -v "${BUILD_DIR}:/dd-trace-py" -e "ARCH=i686" quay.io/pypa/manylinux1_i686 linux32 /bin/bash -c "${build_script}"
+docker pull quay.io/pypa/manylinux1_i686
+docker run -it --rm -v "${BUILD_DIR}:/dd-trace-py" -e "ARCH=i686" quay.io/pypa/manylinux1_i686:latest linux32 /bin/bash -c "${build_script}"
```
Comment (brettlangdon) [inline scripts/build-dist]:
```suggestion
docker pull quay.io/pypa/manylinux1_i686:latest
docker run -it --rm -v "${BUILD_DIR}:/dd-trace-py" -e "ARCH=i686" quay.io/pypa/manylinux1_i686:latest linux32 /bin/bash -c "${build_script}"
```

Comment (brettlangdon) [reply scripts/build-dist]: ?

File: scripts/build-dist (hunk 1 above)
Comment (brettlangdon) [reply scripts/build-dist]: ?

File: scripts/build-dist (hunk 2 above)
Comment (majorgreys) [reply scripts/build-dist]: `docker pull` defaults to the `:latest` tag so adding it here would be redundant

Comment (brettlangdon) [reply scripts/build-dist]: 👍

**Pull request merged**